cEP-0025.md

```diff
@@ -0,0 +1,306 @@
+        for result in dependency_results.get(PyFlakesASTBear.name, []):
+            for node in result.get_nodes(result.module_scope[0],
+                                         FutureImportation):
+                lineno = result.get_node_location(node)[0]
+                corrected = self.remove_import(file, lineno, node.name)
+                yield Result.from_values(
+                    origin=self,
+                    message=('Future import %s found' % node.name),
+                    file=filename,
+                    diffs={filename: corrected},
+                    line=result.get_node_location(node)[0])
+```
+
+## Demonstrating use of PyFlakesASTBear by creating a fixes bear
+
+A hassle free solution to the problem of removing all the future imports
+present in the source code.
+
+Here is a prototype for the NoFutureImportBear that uses PyFlakesASTBear:
+
+```python
+# Imports not mentioned to maintain brevity
+
+class NoFutureImportBear(LocalBear):
+    """
+    Uses pyflakes-enhance-AST to remove future imports
+    """
+    BEAR_DEPS = {PyFlakesASTBear}
+
+    def remove_future_imports(self, file, lineno,
+                              max_col_offset, affected_lines):
+        """
+        Removes all __future__ imports from the input line
+        """
+        diff = Diff(file)
+        line = file[lineno - 1]
+        # If a line contains \ in the end
+        if line.rstrip()[-1] == '\\':
+            next_line = file[lineno]
+            semicolon_index = next_line.find(';')
+            diff.delete_line(lineno)
+            if semicolon_index == -1:
+                diff.delete_line(lineno + 1)
+            else:
+                replacement = next_line[semicolon_index + 1:].lstrip()
+                diff.modify_line(lineno+1, replacement)
+        else:
+            semicolon_indices = [i for i, a in enumerate(line) if a == ';']
+            if len(semicolon_indices) == 0:
+                diff.delete_line(lineno)
+            else:
+                for indice in semicolon_indices:
+                    if indice > max_col_offset:
+                        replacement = line[indice + 1:].lstrip()
+                        diff.modify_line(lineno, replacement)
+                        break
+
+        return diff
+
+    def run(self, filename, file,
+            dependency_results=dict()
+            ):
+        """
+        Uses PyFlakesASTBear to get only FutureImportation
+        """
+        # Preprocess all the lines containing __future__ imports
+        affected_lines = set()
+        # Get the maximum column offset of FutureImportation for a line
+        max_column_for_a_line = dict()
+        for result in dependency_results.get(PyFlakesASTBear.name, []):
+            for node in result.get_nodes(result.module_scope[0],
+                                         FutureImportation):
+                lineno = result.get_node_location(node)[0]
+                col_offset = result.get_node_location(node)[2]
```

Comment:
this `[2]` here isnt self-explanatory.

maybe get_node_location should return something which is easier to use.

Reply:
We could even opt to remove the wrapper function.
`node.source.lineno` or `node.source.col__offset` will be good.